File: backend/src/CLS.Budget.Domain/Entities/AccountCategory.cs
namespace CLS.Budget.Domain.Entities;

/// <summary>
/// Top-level account category (e.g. Credit Card, Loan).
/// System rows have <see cref="TenantId"/> null and <see cref="IsSystem"/> true.
/// Tenant-owned custom categories have a TenantId.
/// </summary>
public class AccountCategory
{
    public int AccountCategoryId { get; set; }
    /// <summary>Null for system-wide seeded categories.</summary>
    public Guid? TenantId { get; set; }
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public bool IsSystem { get; set; }
    public ICollection<AccountSubCategory> SubCategories { get; set; } = [];
}

File: frontend/cls-budget-web/src/features/accounts/types/accountCategory.ts
export interface AccountSubCategoryResponse {
  accountSubCategoryId: number;
  accountCategoryId: number;
  name: string;
  description: string | null;
}

export interface AccountCategoryResponse {
  accountCategoryId: number;
  name: string;
  description: string | null;
  isSystem: boolean;
  subCategories: AccountSubCategoryResponse[];
}

export interface CreateAccountCategoryRequest {
  name: string;
  description?: string | null;
}

export interface CreateAccountSubCategoryRequest {
  accountCategoryId: number;
  name: string;
  description?: string | null;
}
